pkg/webhook/admission/vm-validator.go

```diff
@@ -0,0 +1,482 @@
+		if failure, valid := isValidBallooning(policy); !valid {
+			results = append(results, failure)
+		}
+		if failure, valid := isValidOvercommit(policy); !valid {
+			results = append(results, failure)
+		}
+		if failure, valid := isValidGuaranteed(policy); !valid {
+			results = append(results, failure)
+		}
+	}
+	return results
+}
+
+func isValidBallooning(policy *ovirtsdk.MemoryPolicy) (ValidationFailure, bool) {
+	if ballooning, ok := policy.Ballooning(); ok && ballooning {
+		return ValidationFailure{
+			ID:      VMMemoryPolicyBallooningID,
+			Message: "VM enables memory ballooning",
+		}, false
+	}
+	return ValidationFailure{}, true
+}
+
+func isValidGuaranteed(policy *ovirtsdk.MemoryPolicy) (ValidationFailure, bool) {
+	if guaranteed, ok := policy.Guaranteed(); ok {
+		return ValidationFailure{
+			ID:      VMMemoryPolicyGuaranteedID,
+			Message: fmt.Sprintf("VM specifies guaranteed memory: %d", guaranteed),
+		}, false
+	}
+	return ValidationFailure{}, true
+}
+
+func isValidOvercommit(policy *ovirtsdk.MemoryPolicy) (ValidationFailure, bool) {
+	if overcommit, ok := policy.OverCommit(); ok {
+		if percent, ok := overcommit.Percent(); ok {
+			return ValidationFailure{
+				ID:      VMMemoryPolicyOvercommitPercentID,
+				Message: fmt.Sprintf("VM specifies memory overcommit percent: %d", percent),
+			}, false
+		}
+	}
+	return ValidationFailure{}, true
+}
+
+func isValidMigration(vm *ovirtsdk.Vm) (ValidationFailure, bool) {
+	if migration, ok := vm.Migration(); ok {
+		return ValidationFailure{
+			ID:      VMMigrationID,
+			Message: fmt.Sprintf("VM has migration options specified: %v", migration),
+		}, false
+	}
+	return ValidationFailure{}, true
+}
+
+func isValidMigrationDowntime(vm *ovirtsdk.Vm) (ValidationFailure, bool) {
+	if downtime, ok := vm.MigrationDowntime(); ok {
+		return ValidationFailure{
+			ID:      VMMigrationDowntimeID,
+			Message: fmt.Sprintf("VM has migration downtime secified: %d", downtime),
+		}, false
+	}
+	return ValidationFailure{}, true
+}
+
+func isValidNumaTuneMode(vm *ovirtsdk.Vm) (ValidationFailure, bool) {
+	if mode, ok := vm.NumaTuneMode(); ok {
+		return ValidationFailure{
+			ID:      VMNumaTuneModeID,
+			Message: fmt.Sprintf("VM has NUMA tune mode secified: %s", mode),
+		}, false
+	}
+	return ValidationFailure{}, true
+}
+
+func isValidOrigin(vm *ovirtsdk.Vm) (ValidationFailure, bool) {
+	if origin, ok := vm.Origin(); ok && origin == "kubevirt" {
+		return ValidationFailure{
+			ID:      VMOriginID,
+			Message: "VM has origin set to 'kubevirt'",
+		}, false
+	}
+	return ValidationFailure{}, true
+}
+
+func isValidRandomNumberGeneratorSource(vm *ovirtsdk.Vm) (ValidationFailure, bool) {
+	if rng, ok := vm.RngDevice(); ok {
+		if src, ok := rng.Source(); ok && src != "urandom" {
+			return ValidationFailure{
+				ID:      VMRngDeviceSourceID,
+				Message: fmt.Sprintf("VM has illegal random number generator device source set: %v. Supported value: 'urandom'", src),
```

Comment:
Is it illegal or unsupported?

Reply:
It cannot be mapped in any way - CNV will always use host's urandom if we specify rng option in the spec. Probably "unsupported" would be better.